pkg/disk/disk_windows.go

```diff
@@ -65,40 +68,48 @@ func (m *userDataDiskManager) DetachUserDataDisk() error {
 	return nil
 }
 
-func (m *userDataDiskManager) createDisk(diskPath string) error {
-	size, err := sizeInMB()
-	if err != nil {
-		return fmt.Errorf("failed to get disk size: %w", err)
-	}
+// min_win_disk.zip is a zip directory with a single file (disk.vhdx).
+// disk.vhdx is a 50G max size, sparse, GPT, vhdx file created by diskpart, which contains
+// a single ext4 partition. Since using diskpart requires Administrator privileges,
+// we ship a pre-created disk to be used for the persistent data volume.
+//
+//go:embed min_win_disk.zip
+var minWinDisk []byte
 
-	m.logger.Infof("creating disk at path: %s", diskPath)
-
-	execPath, err := os.Executable()
+func (m *userDataDiskManager) createDisk(diskPath string) error {
+	m.logger.Infof("creating persistent disk: %s", diskPath)
+	r, err := zip.NewReader(bytes.NewReader(minWinDisk), int64(len(minWinDisk)))
 	if err != nil {
-		return fmt.Errorf("failed to get executable path: %w", err)
+		return fmt.Errorf("failed to create zip reader: %w", err)
 	}
 
-	// Put all paths in quotes, since they are being passed to cmd.exe.
-	dpgoPath := fmt.Sprintf(`"%s"`, filepath.Join(string(m.finch), "bin", "dpgo.exe"))
-	diskPathQuoted := fmt.Sprintf(`"%s"`, diskPath)
-	wdPathQuoted := fmt.Sprintf(`"%s"`, filepath.Dir(execPath))
-	sizeStr := fmt.Sprint(size)
-
-	if err := winutil.RunElevated(
-		dpgoPath,
-		wdPathQuoted,
-		[]string{
-			"--json",
-			"--debug",
-			"disk",
-			"create",
-			"--path",
-			diskPathQuoted,
-			"--size",
-			sizeStr,
-		},
-	); err != nil {
-		return fmt.Errorf("failed to run dpgo command: %w", err)
+	for _, f := range r.File {
+		if f.Name == "disk.vhdx" {
```

Comment:
Is there any possibility of the zip archive getting corrupt and not able to find `disk.vhdx`?

Reply:
I don't think so, since its embedded inside finch.exe itself. I can add an error in the case that no file with the name `disk.vhdx` is found though. Or just check if the file exists on disk after the for loop?

Reply:
No I think this is fine, I was just curious about the 204800.